application/app/PathAlgorithmDev.ts

```
@@ -0,0 +1,319 @@
+           { id: "H2.1-MR2", type: "hallway", x: 710, y: 800, floor: "8", adjacent: ["H1-H2.1","stairs-H2.1-2","H2.1-CR2","835"]},
+           { id: "H2.1-CR2", type: "hallway", x: 835, y: 800, floor: "8", adjacent: ["H2.1-MR2","833","837","829","H4-ML"]},
+
+           //Left hallway top to bottom
+           { id: "H3-MU", type: "hallway", x: 185, y: 320, floor: "8", adjacent: ["H2-LC","H3-HE","861"]},
+           { id: "H3-HE", type: "hallway", x: 185, y: 400, floor: "8", adjacent: ["H3-MU","859","H3-M","HE-1"]},
+           { id: "H3-M", type: "hallway", x: 185, y: 500, floor: "8", adjacent: ["H3-HE","H3-ML","857"]},
+           { id: "H3-ML", type: "hallway", x: 185, y: 650, floor: "8", adjacent: ["H3-M","H2.1-LC2","855","853","852","854"]},
+
+           //Right hallway top to bottom
+           { id: "H4-MU", type: "hallway", x: 835, y: 350, floor: "8", adjacent: ["H4-M","H2-CR","889","821","823","820-1"]},
+           { id: "H4-M", type: "hallway", x: 835, y: 500, floor: "8", adjacent: ["H4-ML","H4-MU","820-2","825"]},
+           { id: "H4-ML", type: "hallway", x: 835, y: 650, floor: "8", adjacent: ["H2.1-CR2","H3-M","829","827","822"]},
+
+           //Elevator and stairs hallway
+           { id: "HE-1", type: "hallway", x: 260, y: 400, floor: "8", adjacent: ["H3-HE","stairs-HE", "HE-2", "860"]},
+           { id: "HE-2", type: "hallway", x: 360, y: 400, floor: "8", adjacent: ["HE-1", "H8-elevators", "HE-3", "862", "892"]},
+           { id: "HE-3", type: "hallway", x: 480, y: 400, floor: "8", adjacent: ["HE-2", "H1-HE","840"]},
+
+           { id: "806-01", type: "classroom", x: 490, y: 280, floor: "8", adjacent: ["H1-MU"]},
+           { id: "806-02", type: "classroom", x: 490, y: 318, floor: "8", adjacent: ["H1-MU"]},
+           { id: "806-03", type: "classroom", x: 490, y: 355, floor: "8", adjacent: ["H1-MU"]},
```

The initializeNeighborIfMissing function (line 45) assumes that a missing node type should default to "unknown". Try explicitly handling missing nodes or throwing an error instead of using a default value.